presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -92,14 +92,19 @@ public PlanNode optimize(PlanNode plan, Session session, Map<Symbol, Type> types
         }
 
         Memo memo = new Memo(idAllocator, plan);
+        optimize(memo, session, symbolAllocator, idAllocator);
+
+        return memo.extract();
+    }
+
+    public void optimize(Memo memo, Session session, SymbolAllocator symbolAllocator, PlanNodeIdAllocator idAllocator)
```

Comment:
same comments as in #8852, https://github.com/prestodb/presto/pull/8852/commits/cbf5042c06f6a3592a0a940dda636c776cf3322c#r135787704

Generally, I don't like the idea we have so many concurrent PR into same area of code, because it makes reviewing harder.